src/Databases/DatabaseMemory.cpp

```diff
@@ -95,6 +95,45 @@ void DatabaseMemory::dropTable(
     if (table_uuid != UUIDHelpers::Nil)
         DatabaseCatalog::instance().removeUUIDMappingFinally(table_uuid);
 }
+void DatabaseMemory::dropTableCaseInsensitive(
```

Comment:
I think unifying `dropTableCaseInsensitive` and `dropTable` makes even more sense than in case of `DatabasesCommon::detachTableCase(Insensitive)Unlocked`. The logic is even less trivial, while unifying only requires passing a member function pointer.